index.js

```diff
@@ -32,8 +36,23 @@ function escapeUnsafeChars(unsafeChar) {
     return ESCAPED_CHARS[unsafeChar];
 }
 
+// Escape function body for XSS protection while preserving arrow function syntax
+function escapeFunctionBody(str) {
+    // Escape </script> sequences (case-insensitive) - the main XSS risk
+    // This must be done first before other replacements
+    str = str.replace(SCRIPT_CLOSE_REGEXP, function(match) {
+        return '\\u003C\\u002Fscript\\u003E';
+    });
+    // Also escape </SCRIPT> and other case variations
+    str = str.replace(/<\/SCRIPT>/g, '\\u003C\\u002FSCRIPT\\u003E');
+    // Escape line terminators (these are always unsafe)
+    str = str.replace(/\u2028/g, '\\u2028');
+    str = str.replace(/\u2029/g, '\\u2029');
+    return str;
+}
+
 function generateUID() {
-    var bytes = crypto.getRandomValues(new Uint8Array(UID_LENGTH));
+    var bytes = crypto.randomBytes(UID_LENGTH);
```

Comment:
why did you change it to randomBytes? is it recommended for security reasons?

Reply:
that was some debugging, I reverted it